src/OrleansProviders/SQLServer/CreateTables.sql

```diff
@@ -1,113 +1,1440 @@
+		BEGIN TRANSACTION;
+		UPDATE [OrleansMembershipTable]
+		SET
+			IAmAliveTime = @iAmAliveTime	    
+		WHERE
+			([DeploymentId] = @deploymentId AND @deploymentId IS NOT NULL)
+			AND ([Address] = @address AND @address IS NOT NULL)
+			AND ([Port] = @port AND @port IS NOT NULL)
+			AND ([Generation] = @generation AND @generation IS NOT NULL);
+		COMMIT TRANSACTION;',
+		N''
+	);
+
+	INSERT INTO [OrleansQuery]([Key], [Query], [Description])
+	VALUES
+	(
+		'InsertMembershipVersionKey',
+		N'SET NOCOUNT ON;
+		BEGIN TRANSACTION;
+		INSERT INTO [OrleansMembershipVersionTable]
+		(
+			[DeploymentId],
+			[Timestamp],
+			[Version]	    
+		)
+		SELECT	
+			@deploymentId,
+			SYSUTCDATETIME(),
+			@version        
+		WHERE NOT EXISTS
+		(
+			SELECT 1
+			FROM OrleansMembershipVersionTable
+			WHERE [DeploymentId] = @deploymentId AND @deploymentId IS NOT NULL
+		);
+                                        
+		IF @@ROWCOUNT > 0
+		BEGIN
+			COMMIT TRANSACTION;
+			SELECT CAST(1 AS BIT);
+		END
+		ELSE
+		BEGIN
+			ROLLBACK TRANSACTION;
+			SELECT CAST(0 AS BIT);
+		END',
+		N''
+	);
+
+	INSERT INTO [OrleansQuery]([Key], [Query], [Description])
+	VALUES
+	(
+		'InsertMembershipKey',
+		N'SET NOCOUNT ON;
+		BEGIN TRANSACTION; --  @@TRANCOUNT = 0 -> +1.
+		-- There is no need to check the condition for inserting
+		-- as the necessary condition with regard to table membership
+		-- protocol is enforced as part of the primary key definition.
+		-- Inserting will fail if there is already a membership
+		-- row with the same
+		-- * [DeploymentId] = @deploymentId
+		-- * [Address]		= @address
+		-- * [Port]			= @port
+		-- * [Generation]	= @generation
+		--
+		-- For more information on table membership protocol insert see at
+		-- http://dotnet.github.io/orleans/Runtime-Implementation-Details/Runtime-Tables.html and at
+		-- https://github.com/dotnet/orleans/blob/master/src/Orleans/SystemTargetInterfaces/IMembershipTable.cs
+		INSERT INTO [OrleansMembershipTable]
+		(
+			[DeploymentId],
+			[Address],
+			[Port],
+			[Generation],
+			[HostName],
+			[Status],
+			[ProxyPort],
+			[Primary],
+			[RoleName],
+			[InstanceName],
+			[UpdateZone],
+			[FaultZone],
+			[SuspectingSilos],
+			[SuspectingTimes],
+			[StartTime],
+			[IAmAliveTime]
+		)
+		VALUES
+		(
+			@deploymentId,
+			@address,
+			@port,
+			@generation,
+			@hostName,
+			@status,
+			@proxyPort,
+			@primary,
+			@roleName,
+			@instanceName,
+			@updateZone,
+			@faultZone,
+			@suspectingSilos,
+			@suspectingTimes,
+			@startTime,
+			@iAmAliveTime
+		);
+
+		IF @@ROWCOUNT = 0 ROLLBACK TRANSACTION;
+
+		IF @@TRANCOUNT > 0
+		BEGIN
+			-- The transaction has not been rolled back. The following
+			-- update must succeed or else the whole transaction needs
+			-- to be rolled back.
+			UPDATE [OrleansMembershipVersionTable]
+			SET
+				[Timestamp]	= SYSUTCDATETIME(),
+				[Version]	= @version
+			WHERE
+				([DeploymentId]	= @deploymentId AND @deploymentId IS NOT NULL)
```

Comment:
Having ``NULL`` valued parameters should never happen. Because the behaviour of ``NULL`` in SQL is difficult to predict (in practice one can treat as ``undefined`` in three-valued logic), I chose to be explicit about checking this case. It also documents the queries better, I feel.

Also, one could also test database functionality by providing the queries with parameters and see the results are what is expected and in that case querying would be less hazardous.